Web/src/components/room/ErrorDisplay.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import styled from 'styled-components';
+import { CommError } from './comm/Errors';
+
+const ErrorContainer = styled.div`
+  margin: 10px;
+  background: var(--red);
+  border: white;
+  border-radius: 0.8rem;
+  border-style: dashed;
+`;
+
+const ErrorText = styled.p`
+  color: white;
+  margin: 6px;
+`;
+
+const ErrorTitleText = styled(ErrorText)`
+  font-weight: bold;
+`;
+
+interface ErrorDisplayProps {
+  error: CommError | null;
+  errorDescription: string | null;
+}
+
+interface TitleMessage {
+  title: string;
+  message?: string;
+}
+
+const titleMessageMapping: { [error in CommError]: TitleMessage } = {
```

Comment:
```suggestion
const titleMessageMapping: Record<CommError, TitleMessage> = {
```

Nit: more descriptive type lol

Reply:
https://www.typescriptlang.org/docs/handbook/utility-types.html#recordkt

It’s the same as what you had previously, but I think this looks clearer